// react-date-inputs/dist/stories/ReactDateInputs.stories.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React, { useState } from 'react';
import DateInputs from '../index';
import './styles.css';
export default {
    title: 'React Date Inputs',
    component: DateInputs,
};
var Template = function (args) {
    var _a = useState(args.value), value = _a[0], setValue = _a[1];
    return React.createElement(DateInputs, __assign({}, args, { value: value, onChange: setValue }));
};
export var Simple = Template.bind({});
Simple.argTypes = {
    show: {
        options: ['day', 'month', 'year'],
        control: { type: 'check' },
    },
    dayPlaceholder: {
        control: { type: 'text' },
    },
    monthPlaceholder: {
        control: { type: 'text' },
    },
    yearPlaceholder: {
        control: { type: 'text' },
    },
};
Simple.args = {
    label: 'Date',
    disabled: false,
    show: ['day', 'month', 'year'],
    autoTab: false,
    dayPlaceholder: 'DD',
    monthPlaceholder: 'MM',
    yearPlaceholder: 'YYYY',
};
export var PrePopulated = Template.bind({});
PrePopulated.argTypes = __assign({}, Simple.argTypes);
PrePopulated.args = __assign(__assign({}, Simple.args), { value: new Date(2022, 6, 20) });
export var CustomInputOrder = Template.bind({});
var show = ['year', 'month', 'day'];
CustomInputOrder.argTypes = __assign(__assign({}, Simple.argTypes), { show: show });
CustomInputOrder.args = __assign(__assign({}, Simple.args), { show: show });
export var WithCustomComponents = Template.bind({});
var InputComponent = React.forwardRef(function (props, ref) {
    return React.createElement("input", __assign({ style: props.style }, props, { ref: ref }));
});
InputComponent.displayName = 'InputComponent';
var LabelComponent = function (props) {
    return (React.createElement("label", __assign({ style: {
            textTransform: 'uppercase',
            fontFamily: 'sans-serif',
            fontWeight: 'bold',
            fontSize: '0.8rem',
        } }, props)));
};
WithCustomComponents.argTypes = __assign(__assign({}, Simple.argTypes), { inputComponent: {
        table: { disable: true },
    }, labelComponent: {
        table: { disable: true },
    } });
WithCustomComponents.args = __assign(__assign({}, Simple.args), { inputComponent: InputComponent, labelComponent: LabelComponent, inputComponentProps: {
        style: {
            padding: '6px',
            marginRight: '4px',
            marginTop: '0.2rem',
        },
    } });
